common/src/main/java/com/wynntils/functions/InventoryFunctions.java

```diff
@@ -310,4 +313,25 @@ protected List<String> getAliases() {
             return List.of("held_item", "held_name");
         }
     }
+
+    public static class HeldItemCooldownFunction extends Function<CappedValue> {
+        @Override
+        public CappedValue getValue(FunctionArguments arguments) {
+            Player player = McUtils.player();
+            ItemStack itemStack = player.getItemInHand(InteractionHand.MAIN_HAND);
+            ItemCooldowns cooldowns = player.getCooldowns();
+            ItemCooldowns.CooldownInstance cooldown = cooldowns.cooldowns.get(itemStack.getItem());
+            if (cooldown == null || cooldown.startTime >= cooldown.endTime) return CappedValue.EMPTY; // Sanity check
+
+            int remaining = cooldown.endTime - cooldowns.tickCount;
+            if (remaining <= 0) return CappedValue.EMPTY;
+
+            return new CappedValue(remaining, cooldown.endTime - cooldown.startTime);
```

Comment:
Extract this to the appropriate model, functions should essentially be "public getters" of core components data.